metriq_gym/run.py

```diff
@@ -67,10 +62,11 @@ def poll_job(args: argparse.Namespace, job_manager: JobManager) -> None:
     metriq_job: MetriqGymJob = job_manager.get_job(args.job_id)
     job_type: JobType = JobType(metriq_job.job_type)
     job_data: BenchmarkData = setup_job_data_class(job_type)(**metriq_job.data)
-    job_class = setup_job_class(metriq_job.provider_name)
-    device = setup_device(metriq_job.provider_name, metriq_job.device_name)
-    handler = setup_handler(args, None, job_type)
-    quantum_job = [job_class(job_id, device=device) for job_id in job_data.provider_job_ids]
+    handler = setup_benchmark(args, None, job_type)
+    quantum_job = [
+        load_job(job_id, provider=metriq_job.provider_name, **asdict(job_data))
```

Comment:
Instead of unpacking the dict, do we want to use `serialize` to load the `job_data`?

Reply:
Here we are passing job_data as a dictionary to the kwargs of load_job. Serializing would make job_data into a string, so not a good use here.